docs/manual/docs/user-guide/harvesting/harvesting-database.md

```diff
@@ -0,0 +1,43 @@
+    -   *Group*: Group which owns the harvested records. Only the catalog administrator or users with the profile `UserAdmin` of this group can manage the harvester.
+    -   *User*: User who owns the harvested records.
+
+-   **Schedule**: Scheduling options to execute the harvester. If disabled, the harvester should be executed manually from the harvesters page. If enabled a schedule expression using cron syntax should be configured ([See examples](https://www.quartz-scheduler.org/documentation/quartz-2.1.7/tutorials/crontrigger)).
+
+-   **Configure connection to Database**
+    -   *Server*: The database server IP/Hostname.
+    -   *Port*: The database port. For example, for Postgres usually 5432.
+    -   *Database name*: The database name to connect.
+    -   *Table name*: Table name with the metadata. The name must begin with a letter (a-z) or underscore (_). Subsequent characters in a name can be letters, digits (0-9), or underscores.
+    -   *Metadata field name*: Table field name that contains the metadata. The name must begin with a letter (a-z) or underscore (_). Subsequent characters in a name can be letters, digits (0-9), or underscores.
+    -   *Database type*: Database type. Currently supported Postgres and Oracle.
+    -   *Remote authentication*: Credentials to connect to the database.
+
+-   **Search filter**: allows to define a simple field condition to filter the results.
+    -   *Filter field*: Table field name used to filter the results. The name must begin with a letter (a-z) or underscore (_). Subsequent characters in a name can be letters, digits (0-9), or underscores.
+    -   *Filter value*: Value to filter the results. It can contain wildcards (%).
+    
+-   **Configure response processing for database**
+    -   *Action on UUID collision*: When a harvester finds the same uuid on a record collected by another method (another harvester, importer, dashboard editor,...), should this record be skipped (default), overriden or generate a new UUID?
+    -   *Validate records before import*: If checked, the metadata will be validated after retrieval. If the validation does not pass, the metadata will be skipped. 
+    -   *XSL filter name to apply*: (Optional) The XSL filter is applied to each metadata record. The filter is a process which depends on the metadata schema (see the `process` folder of the metadata schemas).
+
+        It could be composed of parameter which will be sent to XSL transformation using the following syntax: `anonymizer?protocol=MYLOCALNETWORK:FILEPATH&email=gis@organisation.org&thesaurus=MYORGONLYTHEASURUS`
+
+    -   *Batch edits*: (Optional) Allows to update harvested records, using XPATH syntax. It can be used to add, replace or delete element.
+    -   *Translate metadata content*: (Optional) Allows to translate metadata elements. It requires a translation service provider configured in the System settings.
+
+-   **Privileges** - Assign privileges to harvested metadata.
```

Comment:
Maybe you could add a brief description of the structure of the DB to be harvested, in order to avoid having to read the code) ?

Reply:
I can be wrong, as not an expert about this, but `getJdbcTemplate().query` handles that

https://github.com/spring-projects/spring-framework/blob/0da8dee289e4cbe353d4bd6cc0935cd62ab27901/spring-jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java#L627-L634

, it uses this code

https://github.com/spring-projects/spring-framework/blob/0da8dee289e4cbe353d4bd6cc0935cd62ab27901/spring-jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java#L605-L606

That closes the statement and underlying connection.

Reply:
The documentation had already some entries for this:

https://github.com/geonetwork/core-geonetwork/pull/8247/files#diff-b88936cc370d94d9d97a131a54e816da6de85236d379f5470a2b1c769835c8abR24-R25

https://github.com/geonetwork/core-geonetwork/pull/8247/files#diff-b88936cc370d94d9d97a131a54e816da6de85236d379f5470a2b1c769835c8abR29-R31

Please check if requires some improvements or examples.

Reply:
Yes this part is good. I was referring to the part which lists the column types supported which are not mentionned and I came accross them in DatabaseMetadataRetriever.

Reply:
Updated the documentation: https://github.com/geonetwork/core-geonetwork/pull/8247/commits/32c6dcea7547b60723117aeff0e63ae331e96f9e